spec/requests/courses_spec.rb

```diff
@@ -18,24 +18,24 @@
   # Course. As you add validations to Course, be sure to
   # adjust the attributes here as well.
 
-  let!(:user) { User.create(email: "user@test.com", password: "testpass") }
-  let(:subject) { Subject.create!(name: "subject name", user: user) }
-  let(:valid_attributes) { { name: "course name", subject: subject } }
-  let(:invalid_attributes) { { name: nil, subject: subject } }
+  let!(:user) { FactoryBot.create(:user) }
```

Comment:
suggestion: If you want to create user associations and are having trouble with validations like email uniqueness, you might try sequences:
https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#global-sequences